### PR TITLE
*: rename store block to pauseLeaderTransfer

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
The term `block` is ambiguous.

### What is changed and how it works?
Some renames.

### Check List

Tests
- Unit test

Side effects
- Breaking backward compatibility
  - Have API error message change.

### Release note
- No release note

### CODE REVIEW

File: server/core/basic_cluster.go [hunk 1]
```diff
@@ -128,18 +128,18 @@ func (bc *BasicCluster) GetAdjacentRegions(region *RegionInfo) (*RegionInfo, *Re
 	return bc.Regions.GetAdjacentRegions(region)
 }
 
-// BlockStore stops balancer from selecting the store.
-func (bc *BasicCluster) BlockStore(storeID uint64) error {
+// PauseLeaderTransfer stops balancer from selecting the store.
```
Comment (rleungx) [inline server/core/basic_cluster.go:131]: Does the comment need to be changed?

File: server/core/store.go [hunk 2]
```diff
@@ -566,28 +566,28 @@ func (s *StoresInfo) SetStore(store *StoreInfo) {
 	s.stores[store.GetID()] = store
 }
 
-// BlockStore blocks a StoreInfo with storeID.
-func (s *StoresInfo) BlockStore(storeID uint64) errcode.ErrorCode {
-	op := errcode.Op("store.block")
+// PauseLeaderTransfer pauses a StoreInfo with storeID.
```
Comment (rleungx) [inline server/core/store.go:569]: ditto

File: server/core/store.go [hunk 3]
```diff
@@ -566,28 +566,29 @@ func (s *StoresInfo) SetStore(store *StoreInfo) {
...
 }
 
-// UnblockStore unblocks a StoreInfo with storeID.
-func (s *StoresInfo) UnblockStore(storeID uint64) {
+// ResumeLeaderTransfer cleans a store's pause state. The store can be selected
+// as source or target of TransferLeader again.
+func (s *StoresInfo) ResumeLeaderTransfer(storeID uint64) {
 	store, ok := s.stores[storeID]
 	if !ok {
 		log.Fatal("store is unblocked, but it is not found",
```
Comment (lhy1024) [inline server/core/store.go:588]: comment need be updated

Comment (HunDunDM) [reply server/core/store.go:588]: careful!

Comment (disksing) [reply server/core/store.go:588]: good point.

**Pull request merged**